### PR TITLE
Medium: NodeUtilization: Add NodeUtilization agent

### PR DESCRIPTION
**Pull request by krig**
Adds agent which sets node utilization attributes using crm_attribute.

### CODE REVIEW

**Comment by dmuhamedagic** [comment]
On Thu, Feb 02, 2017 at 12:20:20AM -0800, Kristoffer Grönlund wrote:
> Adds agent which sets node utilization attributes using crm_attribute.

There's a ha_pseudo_resource function which provides "management"
of resources which don't have long-running programs. See
ocf-shellfuncs or one of such agents for usage.

It would also be of benefit to use ocf_rarun, it takes care of
the boring details in a correct manner and has some other neat
features. There's documentation at the top of heartbeat/ocf-rarun.

**Comment by krig** [comment]
Thanks for the comments! I rewrote the agent to use `ha_pseudo_resource` and `ocf_rarun`, but I'm not sure if I'm using `ha_pseudo_resource` correctly. Also I need to test that it works, so please don't merge yet :)

**Comment by dmuhamedagic** [comment]
On Fri, Feb 17, 2017 at 12:34:57AM -0800, Kristoffer Grönlund wrote:
> Thanks for the comments! I rewrote the agent to use `ha_pseudo_resource` and `ocf_rarun`, but I'm not sure if I'm using `ha_pseudo_resource` correctly. Also I need to test that it works, so please don't merge yet :)

Great! But we can also upgrade ocf-rarun to do the pseudo
resources too, then just say OCF_PSEUDO_RSC=1. That should be
straightforward.

File: heartbeat/NodeUtilization [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+        $xentool info | awk '/total_memory/{printf("%d\n",$3);exit(0)}'
+    else
+        ocf_log warn "Can only set hv_memory for Xen hypervisor"
+        echo "0"
+    fi
+}
+
+
+set_utilization() {
+    host_name="$(hostname)"
```
Comment (oalbrigt) [inline heartbeat/NodeUtilization]: ocf_local_nodename should be used instead of hostname for it to work on remote nodes and when node name != hostname.

**Comment by krig** [comment]
Switched to using `ocf_local_nodename`, thanks!

File: heartbeat/NodeUtilization [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+#
+#######################################################################
+# Initialization:
+
+: ${OCF_FUNCTIONS_DIR=${OCF_ROOT}/lib/heartbeat}
+. ${OCF_FUNCTIONS_DIR}/ocf-shellfuncs
+
+#######################################################################
+
+meta_data() {
```
Comment (gao-yan) [inline heartbeat/NodeUtilization]: Is this supposed to be NodeUtilization_meta_data()?

**Comment by dmuhamedagic** [comment]
On Mon, Mar 06, 2017 at 09:43:26AM -0800, Gao,Yan wrote:
> gao-yan requested changes on this pull request.
> 
> 
> 
> > +# any, provided herein do not apply to combinations of this program with
> +# other software, or any other product whatsoever.
> +#
> +# You should have received a copy of the GNU General Public License
> +# along with this program; if not, write the Free Software Foundation,
> +# Inc., 59 Temple Place - Suite 330, Boston MA 02111-1307, USA.
> +#
> +#######################################################################
> +# Initialization:
> +
> +: ${OCF_FUNCTIONS_DIR=${OCF_ROOT}/lib/heartbeat}
> +. ${OCF_FUNCTIONS_DIR}/ocf-shellfuncs
> +
> +#######################################################################
> +
> +meta_data() {
> 
> Is this supposed to be NodeUtilization_meta_data()?

Right.

**Comment by krig** [comment]
Yes, good catch!

**Comment by oalbrigt** [comment]
The missing executable makes it fail on "make" (generating the manpage).

I guess we should add that to the CI test as well.

**Comment by oalbrigt** [comment]
Thanks. It's all working fine now.

**Pull request merged**